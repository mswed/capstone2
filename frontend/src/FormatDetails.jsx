import { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { Container, Card, Row, Col } from 'react-bootstrap';
import GrumpyApi from './api';
import Loading from './Loading';
import Checkmark from './Checkmark';

const FormatDetails = () => {
  // Set up state
  const { formatId } = useParams();
  const [formatData, setFormatData] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  const handleCopy = async (value) => {
    try {
      await navigator.clipboard.writeText(value);
    } catch (error) {
      consoel.error('Failed to copy:', error);
    }
  };
  // Fetch format data
  useEffect(() => {
    const getFormatDetails = async () => {
      try {
        const response = await GrumpyApi.getFormatDetails(formatId);
        setFormatData(response);
      } catch (error) {
        console.error('Error fetching camera details', error);
      } finally {
        setIsLoading(false);
      }
    };
    getFormatDetails();
  }, [formatId]);

  if (isLoading) {
    return <Loading />;
  }

  return (
    <Container>
      <Card className="shadow-lg rounded-0 rounded-bottom">
        <Row className="g-0">
          <Col md={3}>
            <Card.Img
              src="/film.png"
              alt="Film strip image"
              className="p-3"
              style={{
                objectFit: 'contain',
                maxHeight: '250px',
                maxWidth: '100%',
              }}
            />
          </Col>
          <Col md={4}>
            <Card.Body className="text-start">
              <Card.Title>
                {formatData.image_format} {formatData.image_aspect} {formatData.format_name}
              </Card.Title>
              <div>
                <dl className="row">
                  <dt className="col-sm-4">Make:</dt>
                  <dd className="col-sm-8">{formatData.make_name}</dd>

                  <dt className="col-sm-4">Model:</dt>
                  <dd className="col-sm-8">{formatData.camera_model}</dd>

                  <dt className="col-sm-4">Resolution:</dt>
                  <dd className="col-sm-8">
                    {formatData.image_width} x {formatData.image_height}
                  </dd>
                  <dt className="col-sm-4">Filmback:</dt>
                  <dd className="col-sm-8">
                    {formatData.sensor_width}mm x {formatData.sensor_height}mm
                  </dd>
                </dl>
              </div>
            </Card.Body>
          </Col>
          <Col md={5} className="text-start p-3">
            <div>
              <dl className="row">
                <dt className="col-sm-4">Anamorphic?:</dt>
                <dd className="col-sm-8">
                  <Checkmark checked={formatData.is_anamorphic} title="anamorphic?" />
                </dd>
                <dt className="col-sm-4">Desqueezed?:</dt>
                <dd className="col-sm-8">
                  <Checkmark checked={formatData.is_dequeezed} title="desqueezed?" />
                </dd>
                <dt className="col-sm-4">Pixel Aspect:</dt>
                <dd className="col-sm-8">{formatData.pixel_aspect}</dd>
                <dt className="col-sm-4">Downsampled?:</dt>
                <dd className="col-sm-8">
                  <Checkmark checked={formatData.is_downsampled} title="downsampled?" />
                </dd>
                <dt className="col-sm-4">Upscaled?:</dt>
                <dd className="col-sm-8">
                  <Checkmark checked={formatData.is_upscaled} title="upscaled?" />
                </dd>
                <dt className="col-sm-4">Codec:</dt>
                <dd className="col-sm-8">{formatData.codec}</dd>
                <dt className="col-sm-4">Raw recording available?:</dt>
                <dd className="col-sm-8">
                  <Checkmark checked={formatData.raw_recording_available} title="row recording available?" />
                </dd>
              </dl>
            </div>
          </Col>
        </Row>
      </Card>
      <Row className="mt-3">
        <Col>
          <Card className="shadow-lg">
            <Card.Body className="text-start">
              <Card.Title>Notes</Card.Title>
              <div>
                <dl>
                  {formatData.notes && <dt className="col-sm-4">Notes:</dt>}
                  {formatData.notes && <dd className="col-sm-8">{formatData.notes}</dd>}
                  {formatData.make_notes && <dt className="col-sm-4">Manufacturer Notes:</dt>}
                  {formatData.make_notes && <dd className="col-sm-8">{formatData.make_notes}</dd>}
                </dl>
              </div>
            </Card.Body>
          </Card>
        </Col>
      </Row>
      <Row className="mt-3">
        <Col md={6}>
          <Card className="shadow-lg">
            <Card.Body className="text-start">
              <Card.Title>3DE Info</Card.Title>
              <div>
                <dl className="row">
                  <dt className="col-sm-4">Filmback Width:</dt>
                  <dd className="col-md-8">
                    {formatData.filmback_width_3de}mm{' '}
                    <i
                      className="bi bi-copy text-muted ms-3"
                      role="button"
                      style={{ cursor: 'pointer' }}
                      aria-label="Copy"
                      onClick={() => handleCopy(formatData.filmback_width_3de)}
                    ></i>
                  </dd>
                  <dt className="col-sm-4">Filmback Height:</dt>
                  <dd className="col-md-8">
                    {formatData.filmback_height_3de}mm{' '}
                    <i
                      className="bi bi-copy text-muted ms-3"
                      role="button"
                      style={{ cursor: 'pointer' }}
                      aria-label="Copy"
                      onClick={() => handleCopy(formatData.filmback_height_3de)}
                    ></i>
                  </dd>
                  <dt className="col-sm-4">Distortion Model:</dt>
                  <dd className="col-sm-8">{formatData.distortion_model_3de}</dd>
                </dl>
              </div>
            </Card.Body>
          </Card>
        </Col>
        <Col md={6}>
          {formatData.tracking_workflow && (
            <Card className="shadow-lg">
              <Card.Body className="text-start">
                <div>
                  <dl className="row">
                    <dt className="col-sm-4">Tracking Workflow:</dt>
                    <dd className="col-sm-8">{formatData.tracking_workflow}</dd>
                  </dl>
                </div>
              </Card.Body>
            </Card>
          )}
        </Col>
      </Row>
    </Container>
  );
};

export default FormatDetails;
