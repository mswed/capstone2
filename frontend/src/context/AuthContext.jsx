import React, { useState, useEffect } from 'react';
import { jwtDecode } from 'jwt-decode';
import GrumpyApi from '../services/api';

// We create the context
const AuthContext = React.createContext();

/**
 * This is a logic only component that wraps our entire app and sets the
 * authentication context and its functions
 *
 * @param {Array}  children - Components to wrap
 * @returns {AuthContext.Provider} - with auth functions for easy access
 */

const AuthProvider = ({ children }) => {
  // We grab stuff from local storage if available
  const storedToken = localStorage.getItem('grumpy-token');
  const storedUser = localStorage.getItem('grumpy-user');

  // We set the token right away so we don't run in to authorization issues
  GrumpyApi.token = storedToken || null;

  // Initialize state
  const [token, setToken] = useState(storedToken);
  const [currentUser, setCurrentuser] = useState(storedUser);

  // Whenever the token changes (login/logout) we need to update
  // both the api module and localStorage
  useEffect(() => {
    if (token && token.trim !== '') {
      // We only set the token if it's not empty

      GrumpyApi.token = token;
      localStorage.setItem('grumpy-token', token || '');
      localStorage.setItem('grumpy-user', currentUser || '');
    } else {
      GrumpyApi.token = null;
      localStorage.removeItem('grumpy-token');
      localStorage.removeItem('grumpy-user');
    }
  }, [token]);

  // TODO:
  // // We also want to get the user favorites here
  // useEffect(() => {
  //   if (!token) return;
  //   async function getUserData() {
  //     const userData = await GrumpuyApi.getUser(currentUser);
  //     setFavorites(new Set(userData.favorites));
  //   }
  //   getUserData();
  // }, [token]);

  const login = async (username, password) => {
    try {
      const authToken = await GrumpyApi.login(username, password);
      setToken(authToken);
      const decoded = jwtDecode(authToken);
      setCurrentuser(decoded.username);
      return { success: true };
    } catch (error) {
      return { success: false, error };
    }
  };

  const register = async (username, password, firstName, lastName, email) => {
    try {
      const token = await GrumpyApi.register(username, password, firstName, lastName, email);
      logout();
      return { success: true };
    } catch (error) {
      return { success: false, error };
    }
  };

  const logout = () => {
    // TODO: Do I need to logout of the backend here too?
    setToken('');
  };

  // TODO: We want to add favorites here
  // const addFavorite = async (jobId) => {
  //   const res = await GrumpuyApi.favoriteFormat(currentUser, jobId);
  //   if (res) {
  //     setFavorites((currentFavorites) => currentFavorites.add(res.favorites));
  //   }
  // };
  //

  // TODO: get items that have already been favorited
  // const alreadyFavorited = (formatId) => {
  //   return favFormats.has(formatId);
  // };

  // Create the value object
  const value = {
    token,
    currentUser,
    login,
    register,
    logout,
  };

  // Wrap everything in the context
  return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>;
};

export { AuthContext, AuthProvider };
