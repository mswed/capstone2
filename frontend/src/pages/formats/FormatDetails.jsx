import { useState, useEffect, useContext } from 'react';
import { useParams } from 'react-router-dom';
import { Container, Card, Row, Col } from 'react-bootstrap';
import GrumpyApi from '../../services/api';
import Loading from '../../components/ui/Loading';
import Checkmark from '../../components/ui/Checkmark';
import FormatForm from '../../components/forms/FormatForm.jsx';
import ActionBar from '../../components/ui/ActionBar.jsx';
import ConfirmDialog from '../../components/ui/ConfirmDialog.jsx';
import ModalWindow from '../../components/ui/ModalWindow.jsx';
import { AuthContext } from '../../context/AuthContext.jsx';
import { useNavigate } from 'react-router-dom';
import useSources from '../../hooks/useSources.js';

const FormatDetails = () => {
  const navigate = useNavigate();

  // Set up state
  const { formatId } = useParams();
  const [formatData, setFormatData] = useState([]);
  const [isLoading, setIsLoading] = useState(true);
  const [showEditFormatModal, setShowEditFormatModal] = useState(false);
  const [showConfirmDelete, setConfirmDelete] = useState(false);
  const { sources, setSources } = useSources();

  const { token } = useContext(AuthContext);

  const actionButtons = [
    {
      text: 'Delete Format',
      variant: 'outline-danger',
      onClick: () => setConfirmDelete(true),
    },
    {
      text: 'Edit Format',
      variant: 'outline-warning',
      onClick: () => setShowEditFormatModal(true),
    },
  ];

  const handleCopy = async (value) => {
    try {
      await navigator.clipboard.writeText(value);
    } catch (error) {
      consoel.error('Failed to copy:', error);
    }
  };

  const handleUpdateFormat = async (formatData) => {
    try {
      const updatedFormatDetails = await GrumpyApi.updateFormat(formatId, formatData);
      setFormatData((prev) => ({ ...prev, ...updatedFormatDetails }));

      // Close the modal
      setShowEditFormatModal(false);
    } catch (error) {
      console.error('Failed to update format:', error);
    }
  };

  const handleAddSource = async (sourceData) => {
    try {
      const newSource = await GrumpyApi.addSource(sourceData);
      setSources((prev) => [...prev, newSource]);

      // Retrun the new source so we can update the UI
      return newSource;
    } catch (error) {
      console.error('Failed to add source', error);

      throw error;
    }
  };

  const handleDeleteFormat = async () => {
    const response = await GrumpyApi.deleteFormat(formatId);
    if (response.success) {
      navigate(-1);
    }
  };

  // Fetch format data
  useEffect(() => {
    const getFormatDetails = async () => {
      try {
        const response = await GrumpyApi.getFormatDetails(formatId);
        setFormatData(response);
      } catch (error) {
        console.error('Error fetching camera details', error);
      } finally {
        setIsLoading(false);
      }
    };
    getFormatDetails();
  }, [formatId]);

  if (isLoading) {
    return <Loading />;
  }

  return (
    <Container>
      <ConfirmDialog
        show={showConfirmDelete}
        title={`Delete format "${formatData.imageFormat} ${formatData.imageAspect} ${formatData.formatName}"`}
        message={'Are you sure?'}
        confirmText="Delete"
        onConfirm={handleDeleteFormat}
        onCancel={() => setConfirmDelete(false)}
      />
      <Card className="shadow-lg rounded-0 rounded-bottom">
        <Row className="g-0">
          <Col md={3}>
            <Card.Img
              src="/film.png"
              alt="Film strip image"
              className="p-3"
              style={{
                objectFit: 'contain',
                maxHeight: '250px',
                maxWidth: '100%',
              }}
            />
          </Col>
          <Col md={4}>
            <Card.Body className="text-start">
              <Card.Title>
                {formatData.imageFormat} {formatData.imageAspect} {formatData.formatName}
              </Card.Title>
              <div>
                <dl className="row">
                  <dt className="col-sm-4">Make:</dt>
                  <dd className="col-sm-8">{formatData.makeName}</dd>

                  <dt className="col-sm-4">Model:</dt>
                  <dd className="col-sm-8">{formatData.cameraModel}</dd>

                  <dt className="col-sm-4">Resolution:</dt>
                  <dd className="col-sm-8">
                    {formatData.imageWidth} x {formatData.imageHeight}
                  </dd>
                  <dt className="col-sm-4">Filmback:</dt>
                  <dd className="col-sm-8">
                    {formatData.sensorWidth}mm x {formatData.sensorHeight}mm
                  </dd>
                </dl>
              </div>
            </Card.Body>
          </Col>
          <Col md={5} className="text-start p-3">
            <div>
              <dl className="row">
                <dt className="col-sm-4">Anamorphic?:</dt>
                <dd className="col-sm-8">
                  <Checkmark checked={formatData.isAnamorphic} title="anamorphic?" />
                </dd>
                <dt className="col-sm-4">Desqueezed?:</dt>
                <dd className="col-sm-8">
                  <Checkmark checked={formatData.isDesqueezed} title="desqueezed?" />
                </dd>
                <dt className="col-sm-4">Pixel Aspect:</dt>
                <dd className="col-sm-8">{formatData.pixelAspect}</dd>
                <dt className="col-sm-4">Downsampled?:</dt>
                <dd className="col-sm-8">
                  <Checkmark checked={formatData.isDownsampled} title="downsampled?" />
                </dd>
                <dt className="col-sm-4">Upscaled?:</dt>
                <dd className="col-sm-8">
                  <Checkmark checked={formatData.isUpscaled} title="upscaled?" />
                </dd>
                <dt className="col-sm-4">Codec:</dt>
                <dd className="col-sm-8">{formatData.codec}</dd>
                <dt className="col-sm-4">Raw recording available?:</dt>
                <dd className="col-sm-8">
                  <Checkmark checked={formatData.rawRecordingAvailable} title="row recording available?" />
                </dd>
              </dl>
            </div>
          </Col>
        </Row>
      </Card>
      <ModalWindow
        show={showEditFormatModal}
        onHide={() => setShowEditFormatModal(false)}
        title={`Edit format ${formatData.imageFormat} ${formatData.imageAspect} ${formatData.formatName}`}
        form={<FormatForm onSubmit={handleUpdateFormat} formatData={formatData} sources={sources} onSourceAdded={handleAddSource} buttonLabel="Update" />}
      />
      {token && <ActionBar buttons={actionButtons} className="mt-3" />}
      <Row className="mt-3">
        <Col>
          <Card className="shadow-lg">
            <Card.Body className="text-start">
              <Card.Title>Notes</Card.Title>
              <div>
                <dl>
                  {formatData.notes && <dt className="col-sm-4">Notes:</dt>}
                  {formatData.notes && <dd className="col-sm-8">{formatData.notes}</dd>}
                  {formatData.make_notes && <dt className="col-sm-4">Manufacturer Notes:</dt>}
                  {formatData.make_notes && <dd className="col-sm-8">{formatData.makeNotes}</dd>}
                </dl>
              </div>
            </Card.Body>
          </Card>
        </Col>
      </Row>
      <Row className="mt-3">
        <Col md={6}>
          <Card className="shadow-lg">
            <Card.Body className="text-start">
              <Card.Title>3DE Info</Card.Title>
              <div>
                <dl className="row">
                  <dt className="col-sm-4">Filmback Width:</dt>
                  <dd className="col-md-8">
                    {formatData.filmbackWidth3de}mm{' '}
                    <i
                      className="bi bi-copy text-muted ms-3"
                      role="button"
                      style={{ cursor: 'pointer' }}
                      aria-label="Copy"
                      onClick={() => handleCopy(formatData.filmbackWidth3de)}
                    ></i>
                  </dd>
                  <dt className="col-sm-4">Filmback Height:</dt>
                  <dd className="col-md-8">
                    {formatData.filmbackHeight3de}mm{' '}
                    <i
                      className="bi bi-copy text-muted ms-3"
                      role="button"
                      style={{ cursor: 'pointer' }}
                      aria-label="Copy"
                      onClick={() => handleCopy(formatData.filmbackHeight3de)}
                    ></i>
                  </dd>
                  <dt className="col-sm-4">Distortion Model:</dt>
                  <dd className="col-sm-8">{formatData.distortionModel3de}</dd>
                </dl>
              </div>
            </Card.Body>
          </Card>
        </Col>
        <Col md={6}>
          {formatData.tracking_workflow && (
            <Card className="shadow-lg">
              <Card.Body className="text-start">
                <div>
                  <dl className="row">
                    <dt className="col-sm-4">Tracking Workflow:</dt>
                    <dd className="col-sm-8">{formatData.trackingWorkflow}</dd>
                  </dl>
                </div>
              </Card.Body>
            </Card>
          )}
        </Col>
      </Row>
    </Container>
  );
};

export default FormatDetails;
