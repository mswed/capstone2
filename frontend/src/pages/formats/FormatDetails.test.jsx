import { vi, it, expect, beforeEach } from 'vitest';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import FormatDetails from './FormatDetails.jsx';
import { AuthContext } from '../../context/AuthContext.jsx';
import { MessagesProvider } from '../../context/MessageContext.jsx';
import GrumpyApi from '../../services/api.js';

// Mock API calls
vi.mock('../../services/api', () => ({
  default: {
    getFormatDetails: vi.fn(() =>
      Promise.resolve({
        id: 1,
        camera: 1,
        cameraModel: 'Alexa 35',
        makeName: 'ARRI',
        imageFormat: '4.6K',
        imageAspect: '3:2',
        formatName: 'Open Gate',
        sensorWidth: 28.0,
        sensorHeight: 19.2,
        imageWidth: 4608,
        imageHeight: 3164,
        pixelAspect: 1.0,
        isAnamorphic: false,
        isDesqueezed: false,
        anamorphicSqueeze: null,
        filmbackWidth3de: 28.0,
        filmbackHeight3de: 19.2,
        isDownsampled: false,
        isUpscaed: false,
        codec: 'ARRIRAW',
        rawReordingAvailable: true,
        source: 1,
        notes: 'sample notes',
        makeNotes:
          '4.6K 3:2 Open Gate provides maximum image quality, resolution, and flexibility in post for many spherical and anamorphic lenses in an image area slightly larger than traditional Super 35 film specifications.',
        trackingWorkflow: 'Do this then that',
      })
    ),
    updateFormat: vi.fn(() =>
      Promise.resolve({
        success: 'Partialy updated format',
        format: {
          id: 1,
          camera: 1,
          cameraModel: 'Alexa 35 Updated',
          makeName: 'ARRI',
          imageFormat: '4.6K',
          imageAspect: '3:2',
          formatName: 'Open Gate',
          sensorWidth: 28.0,
          sensorHeight: 19.2,
          imageWidth: 4608,
          imageHeight: 3164,
          pixelAspect: 1.0,
          isAnamorphic: false,
          isDesqueezed: false,
          anamorphicSqueeze: null,
          filmbackWidth3de: 28.0,
          filmbackHeight3de: 19.2,
          isDownsampled: false,
          isUpscaed: false,
          codec: 'ARRIRAW',
          rawReordingAvailable: true,
          source: 1,
          notes: 'sample notes',
          makeNotes:
            '4.6K 3:2 Open Gate provides maximum image quality, resolution, and flexibility in post for many spherical and anamorphic lenses in an image area slightly larger than traditional Super 35 film specifications.',
          trackingWorkflow: 'Do this then that',
        },
      })
    ),
    deleteFormat: vi.fn(() =>
      Promise.resolve({
        success: 'Format deleted',
      })
    ),
    getSources: vi.fn(() =>
      Promise.resolve([
        {
          id: 1,
          name: 'Sample Source',
          url: 'http://www.samplesource.com',
          fileName: 'sample.pdf',
          note: 'a fake source',
        },
      ])
    ),
  },
}));

describe('FormatDetails component', () => {
  /**
   * Render the componenet with authentication context
   *
   * @returns {Component} - Makes component wrapped in MemoryRouter and AuthContext
   */

  const renderWithAuth = (authValue = { token: 'fake-token', currentUser: 'testUser' }) => {
    return render(
      <MemoryRouter initialEntries={['/formats/1']}>
        <MessagesProvider>
          <AuthContext.Provider value={authValue}>
            <Routes>
              <Route path="/formats/:formatId" element={<FormatDetails />} />
            </Routes>
          </AuthContext.Provider>
        </MessagesProvider>
      </MemoryRouter>
    );
  };

  beforeEach(() => {
    // Clear mocks
    vi.clearAllMocks();
  });

  it('renders without crashing', async () => {
    renderWithAuth();
    await screen.findByText('Alexa 35');
  });

  it('matches snapshot', async () => {
    const { asFragment } = renderWithAuth();

    await screen.findByText('Alexa 35');
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the correct info', async () => {
    renderWithAuth();

    await waitFor(() => {
      expect(screen.getByText('ARRI')).toBeInTheDocument();
      expect(screen.getByText('Alexa 35')).toBeInTheDocument();
      expect(screen.getByText('4608 x 3164')).toBeInTheDocument();
      expect(screen.getByText('28mm x 19.2mm')).toBeInTheDocument();
      expect(screen.getByText('ARRIRAW')).toBeInTheDocument();
      expect(screen.getByText('sample notes')).toBeInTheDocument();
      expect(
        screen.getByText(
          '4.6K 3:2 Open Gate provides maximum image quality, resolution, and flexibility in post for many spherical and anamorphic lenses in an image area slightly larger than traditional Super 35 film specifications.'
        )
      ).toBeInTheDocument();
    });
  });

  it('shows action bar for admin', async () => {
    renderWithAuth({
      token: 'fake-token',
      currentUser: 'admminUser',
      isAdmin: true,
    });
    await screen.findByText('ARRI');
    expect(screen.getByText('Edit Format')).toBeInTheDocument();
  });

  it('hides action bar for regular users', async () => {
    renderWithAuth({ token: 'fake-token', currentUser: 'admminUser' });
    await screen.findByText('ARRI');
    expect(screen.queryByText('Edit Format')).not.toBeInTheDocument();
  });

  it('can edit the format when admin', async () => {
    renderWithAuth({
      token: 'fake-token',
      currentUser: 'adminUser',
      isAdmin: true,
    });
    await screen.findByText('ARRI');

    // Open modal
    fireEvent.click(screen.getByText('Edit Format'));
    expect(screen.getByText('Edit format 4.6K 3:2 Open Gate')).toBeInTheDocument();

    // Fill out the required fields
    const imageFormatInput = screen.getByLabelText(/image format/i);

    fireEvent.change(imageFormatInput, { target: { value: 'Updated Format' } });

    // Submit the form
    fireEvent.click(screen.getByText('Update'));

    // Modal should close after successful submission
    await waitFor(() => {
      expect(screen.queryByText('Edit format 4.6K 3:2 Open Gate')).not.toBeInTheDocument();
    });
  });

  it('can delete the format when admin', async () => {
    renderWithAuth({
      token: 'fake-token',
      currentUser: 'adminUser',
      isAdmin: true,
    });
    await screen.findByText('ARRI');

    // Open modal
    fireEvent.click(screen.getByText('Delete Format'));
    expect(screen.getByText('Delete format "4.6K 3:2 Open Gate"')).toBeInTheDocument();

    // Click the button
    fireEvent.click(screen.getByText('Delete'));

    // Modal should close after successful submission
    await waitFor(() => {
      expect(GrumpyApi.deleteFormat).toHaveBeenCalledWith('1');
    });
  });
});
