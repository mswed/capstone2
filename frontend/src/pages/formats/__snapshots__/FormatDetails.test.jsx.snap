// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormatDetails component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="shadow-lg rounded-0 rounded-bottom card"
    >
      <div
        class="g-0 row"
      >
        <div
          class="col-md-3"
        >
          <img
            alt="Film strip image"
            class="card-img p-3"
            src="/film.png"
            style="object-fit: contain; max-height: 250px; max-width: 100%;"
          />
        </div>
        <div
          class="col-md-4"
        >
          <div
            class="text-start card-body"
          >
            <div
              class="card-title h5"
            >
              4.6K 3:2 Open Gate
            </div>
            <div>
              <dl
                class="row"
              >
                <dt
                  class="col-sm-4"
                >
                  Make:
                </dt>
                <dd
                  class="col-sm-8"
                >
                  ARRI
                </dd>
                <dt
                  class="col-sm-4"
                >
                  Model:
                </dt>
                <dd
                  class="col-sm-8"
                >
                  Alexa 35
                </dd>
                <dt
                  class="col-sm-4"
                >
                  Resolution:
                </dt>
                <dd
                  class="col-sm-8"
                >
                  4608 x 3164
                </dd>
                <dt
                  class="col-sm-4"
                >
                  Filmback:
                </dt>
                <dd
                  class="col-sm-8"
                >
                  28mm x 19.2mm
                </dd>
              </dl>
            </div>
          </div>
        </div>
        <div
          class="text-start p-3 col-md-5"
        >
          <div>
            <dl
              class="row"
            >
              <dt
                class="col-sm-4"
              >
                Anamorphic?:
              </dt>
              <dd
                class="col-sm-8"
              >
                <span
                  class="text-danger"
                >
                  ✗
                </span>
              </dd>
              <dt
                class="col-sm-4"
              >
                Desqueezed?:
              </dt>
              <dd
                class="col-sm-8"
              >
                <span
                  class="text-danger"
                >
                  ✗
                </span>
              </dd>
              <dt
                class="col-sm-4"
              >
                Pixel Aspect:
              </dt>
              <dd
                class="col-sm-8"
              >
                1
              </dd>
              <dt
                class="col-sm-4"
              >
                Downsampled?:
              </dt>
              <dd
                class="col-sm-8"
              >
                <span
                  class="text-danger"
                >
                  ✗
                </span>
              </dd>
              <dt
                class="col-sm-4"
              >
                Upscaled?:
              </dt>
              <dd
                class="col-sm-8"
              >
                <span
                  class="text-danger"
                >
                  ✗
                </span>
              </dd>
              <dt
                class="col-sm-4"
              >
                Codec:
              </dt>
              <dd
                class="col-sm-8"
              >
                ARRIRAW
              </dd>
              <dt
                class="col-sm-4"
              >
                Raw recording available?:
              </dt>
              <dd
                class="col-sm-8"
              >
                <span
                  class="text-danger"
                >
                  ✗
                </span>
              </dd>
            </dl>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mt-3 row"
    >
      <div
        class="col"
      >
        <div
          class="shadow-lg card"
        >
          <div
            class="text-start card-body"
          >
            <div
              class="card-title h5"
            >
              Notes
            </div>
            <div>
              <dl>
                <dt
                  class="col-sm-4"
                >
                  Notes:
                </dt>
                <dd
                  class="col-sm-8"
                >
                  sample notes
                </dd>
                <dt
                  class="col-sm-4"
                >
                  Manufacturer Notes:
                </dt>
                <dd
                  class="col-sm-8"
                >
                  4.6K 3:2 Open Gate provides maximum image quality, resolution, and flexibility in post for many spherical and anamorphic lenses in an image area slightly larger than traditional Super 35 film specifications.
                </dd>
              </dl>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mt-3 row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="shadow-lg card"
        >
          <div
            class="text-start card-body"
          >
            <div
              class="card-title h5"
            >
              3DE Info
            </div>
            <div>
              <dl
                class="row"
              >
                <dt
                  class="col-sm-4"
                >
                  Filmback Width:
                </dt>
                <dd
                  class="col-md-8"
                >
                  28mm 
                  <i
                    aria-label="Copy"
                    class="bi bi-copy text-muted ms-3"
                    role="button"
                    style="cursor: pointer;"
                  />
                </dd>
                <dt
                  class="col-sm-4"
                >
                  Filmback Height:
                </dt>
                <dd
                  class="col-md-8"
                >
                  19.2mm 
                  <i
                    aria-label="Copy"
                    class="bi bi-copy text-muted ms-3"
                    role="button"
                    style="cursor: pointer;"
                  />
                </dd>
                <dt
                  class="col-sm-4"
                >
                  Distortion Model:
                </dt>
                <dd
                  class="col-sm-8"
                />
              </dl>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-md-6"
      />
    </div>
  </div>
</DocumentFragment>
`;
